osu.Game/Screens/Select/Leaderboards/DrawableRank.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Modes;
+using osu.Framework.Extensions;
+
+namespace osu.Game.Screens.Select.Leaderboards
+{
+    public class DrawableRank : Container
+    {
+        private Sprite sprite;
+        private TextureStore textures;
+
+        private ScoreRank rank;
+        public ScoreRank Rank
+        {
+            get { return rank; }
+            set
+            {
+                if (value == rank) return;
+                rank = value;
+                sprite.Texture = textures.Get($@"Badges/ScoreRanks/{rank.GetDescription()}");

```

Comment:
don't duplicate code. make a `updateTexture` method. also this will nullref if called before `load` is run.

- make an `updateTexture` method and call from here.
- inside `updateTexture` add an `if (!IsLoaded) return` to ensure it's not run more than once (or run before texturestore is available).
- move the call in `load` to be in `LoadComplete`. If you leave it in load there's potential thread race scenarios if the Rank is changed while async loading is occurring.

An alternative would be to not allow `Rank` to be settable. Might be the easier way out if that functionality is not yet required.